flask/app/genes.py

```diff
@@ -0,0 +1,104 @@
+from csv import DictWriter, QUOTE_MINIMAL
+from io import StringIO
+import traceback
+from flask import abort, jsonify, request, Blueprint, Response, send_file
+from flask_login import login_required
+from sqlalchemy import func
+from .extensions import db
+from .models import Gene
+from .utils import paged
+
+genes_blueprint = Blueprint(
+    "genes",
+    __name__,
+)
+
+
+@genes_blueprint.route("/api/summary/genes", methods=["GET"])
+@paged
+@login_required
+def genes(page, limit):
+    """ index route for genes """
+    search = request.args.get("search", type=str)
+    gene_query = db.session.query(Gene)
+    if search:
+        gene_query = gene_query.filter(func.instr(Gene.hgnc_gene_name, search))
+
+    total_count = gene_query.with_entities(
+        func.count(func.distinct(Gene.gene_id))
+    ).scalar()
+    if limit and page:
+        gene_query.limit(limit).offset(page * limit)
+    results = gene_query.all()
```

Comment:
You can use the established one-line syntax `.limit(limit).offset(page * (limit or 0)).all()`.

Maybe I should leave in a comment why it works:
- `limit` is either positive or None. `.limit(None)` is a safe no-op
- `page` is non-negative. Requesting page 0 with no limit is also the same as without

Reply:
updated in [616323a](https://github.com/ccmbioinfo/stager/commit/616323a7b787eef2dd3f960032ee95a1cf415d5f#diff-74bfb2f82124f4d2a0aeb38cdff367fb949bf5e0542ceb4dbebe22e7241caf72R31)